flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/format/single/SingleValueRowSerializer.java

```diff
@@ -0,0 +1,120 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.format.single;
+
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.api.common.serialization.SerializationSchema;
+import org.apache.flink.api.common.typeinfo.BasicTypeInfo;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeutils.base.BooleanSerializer;
+import org.apache.flink.api.common.typeutils.base.ByteSerializer;
+import org.apache.flink.api.common.typeutils.base.DoubleSerializer;
+import org.apache.flink.api.common.typeutils.base.FloatSerializer;
+import org.apache.flink.api.common.typeutils.base.IntSerializer;
+import org.apache.flink.api.common.typeutils.base.LongSerializer;
+import org.apache.flink.api.common.typeutils.base.ShortSerializer;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.RowData.FieldGetter;
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.LogicalTypeRoot;
+import org.apache.flink.table.types.logical.RowType;
+
+import java.io.IOException;
+
+/**
+ * SingleValueSerializer: deserialize or serialize single value.
+ */
+public class SingleValueRowSerializer implements DeserializationSchema<RowData>,
+	SerializationSchema<RowData> {
+
+	private TypeInformation<RowData> typeInfo;
+	private LogicalType singleValueLogicalType;
+	private SingleValueSerializer singleValueSerializer;
+	private FieldGetter fieldGetter;
+
+	public SingleValueRowSerializer(RowType rowType,
+		TypeInformation<RowData> resultTypeInfo) {
+		this.typeInfo = resultTypeInfo;
+		this.singleValueLogicalType = rowType.getTypeAt(0);
+		this.fieldGetter = RowData.createFieldGetter(singleValueLogicalType, 0);
+		this.singleValueSerializer = initSingleValueSerializer(singleValueLogicalType.getTypeRoot());
+	}
+
+	@Override
+	public RowData deserialize(byte[] message) throws IOException {
+		GenericRowData genericRowData = new GenericRowData(1);
+		genericRowData.setField(0, singleValueSerializer.deserialize(message));
+		return genericRowData;
+	}
+
+	@Override
+	public boolean isEndOfStream(RowData nextElement) {
+		return false;
+	}
+
+	@Override
+	public TypeInformation<RowData> getProducedType() {
+		return typeInfo;
+	}
+
+	@Override
+	public byte[] serialize(RowData element) {
+		return singleValueSerializer.serialize(fieldGetter.getFieldOrNull(element));
+	}
+
+	private SingleValueSerializer initSingleValueSerializer(LogicalTypeRoot typeRoot) {
+		switch (typeRoot) {
+			case CHAR:
+			case VARCHAR:
```

Comment:
Considering this serializer only works here, I think it's not suitable to add new public class here, such as `BasicSingleValueSerializer`.  
You can add a private function interface such as `SerializationRuntimeConverter` and expose `serialize` method. Therefore, you can use lambda function here instead of adding new class. 
You can take a look at `JsonRowDataSerializationSchema` .